modules/geo/src/internalClusterTest/java/org/opensearch/geo/GeoPluginIntegTestCase.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.geo;
+
+import org.opensearch.index.mapper.GeoShapeFieldMapper;
+import org.opensearch.plugins.Plugin;
+import org.opensearch.test.OpenSearchIntegTestCase;
+import org.opensearch.test.TestGeoShapeFieldMapperPlugin;
+
+import java.util.Collection;
+import java.util.Collections;
+
+/**
+ * This is the base class for all the Geo related integration tests. Use this class to add the features and settings
+ * for the test cluster on which integration tests are running.
+ */
+public abstract class GeoPluginIntegTestCase extends OpenSearchIntegTestCase {
+    /**
+     * Returns a collection of plugins that should be loaded on each node for doing the integration tests. As this
+     * geo plugin is not getting packaged in a zip, we need to load it before the tests run.
+     *
+     * @return List of {@link Plugin}
+     */
+    @Override
+    protected Collection<Class<? extends Plugin>> nodePlugins() {
+        return Collections.singletonList(GeoPlugin.class);
+    }
+
+    /**
+     * This was added as a backdoor to Mock the implementation of {@link GeoShapeFieldMapper} which was coming from
+     * {@link GeoPlugin}. Mock implementation is {@link TestGeoShapeFieldMapperPlugin}. Now we are using the
+     * {@link GeoPlugin} in our integration tests we need to override this functionality to avoid multiple mapper
+     * error.
+     *
+     * @return boolean
+     */
+    @Override
+    protected boolean addMockGeoShapeFieldMapper() {
+        return false;
```

Comment:
In what test scenarios would this get set to `true`?

Reply:
As per the documentation there are test cases which are written in the server folder which depends on this, which test the indexing of geoshape as the geoshape mapper is still in the server folder. But on running this code without this makes the result successful. Let me just update the PR by removing these functions.